frontend/public/lib/getHubData.ts

```diff
@@ -0,0 +1,122 @@
+import { join } from "path";
+import { LinkedHub } from "../../src/types";
+import { apiRequest } from "./apiOperations";
+
+const getHubData = async (url_slug, locale) => {
+  try {
+    const resp = await apiRequest({
+      method: "get",
+      url: `/api/hubs/${url_slug}/`,
+      locale: locale,
+    });
+    return resp.data;
+  } catch (err: any) {
+    if (err.response && err.response.data) {
+      console.log(err.response.data);
+      console.error("Error in getHubData!: " + err.response.data.detail);
+    }
+    return null;
+  }
+};
+
+const getHubAmbassadorData = async (url_slug, locale) => {
+  try {
+    const resp = await apiRequest({
+      method: "get",
+      url: `/api/hubs/${url_slug}/ambassador/`,
+      locale: locale,
+    });
+    return resp.data;
+  } catch (err: any) {
+    if (err.response && err.response.data)
+      console.log("Error in getHubAmbassadorData: " + err.response.data.detail);
+    console.log(err);
+    return null;
+  }
+};
+const getHubSupportersData = async (url_slug, locale) => {
+  try {
+    const resp = await apiRequest({
+      method: "get",
+      url: `/api/hubs/${url_slug}/supporters/`,
+      locale: locale,
+    });
+    return resp.data;
+  } catch (err: any) {
+    //Don't log an error if there simply are no supporters for this hub
+    if (err?.response?.status === 404) {
+      return null;
+    }
+    if (err.response && err.response.data)
+      console.log("Error in getHubSupportersData: " + err.response.data.detail);
+    console.log(err);
+    return null;
+  }
+};
+
+function parseLinkedHubData(data): LinkedHub | null {
+  if (!data || !data.name || !data.url_slug || !data.icon) {
+    console.error("Invalid linked hub data:", data);
+    return null;
+  }
+
+  const url_segments = data.url_slug.split("_");
+  const url_segment = join(...url_segments);
+  console.log("url_segment", url_segment);
+
+  return {
+    hubName: data.name,
+    hubUrl: `/hubs/${url_segment}/browse`,
+    icon: data.icon,
+    backgroundColor: data.icon_background_color,
+  };
+}
```

Comment:
Parses LinkedHub data to objects of `LinkedHub`. As the hub url of a sub hub stored in the backend does not represent the path used in the frontend, this has to be handled here, for example: 
- backend: `perth_energy`
- frontend `perth/energy/browse` instead of `perth_energy/browse`